tests/test_data.py

```diff
@@ -117,3 +117,6 @@ def test_exported_database_includes_headers(self):
             reader = csv.reader(f, delimiter=',')
             header = next(reader)
             assert "creation_time" in header
+
+    def test_archive_data_error(self):
+        assert IOError, dallinger.data.archive_data("12345-6789")
```

Comment:
I don't think this does what you think it does. It asserts that `IOError` is truthy, and if it's not then it shows an assertion error with the result of `dallinger.data.archive_data("12345-6789")` as an explanatory message. In other words this form of assert is equivalent to `if not IOError: raise AssertionError(dallinger.data.archive_data("12345-6789"))`

I think you probably want to call `archive_data`, then assert that the destination file is where it should be, and then delete the destination file.